bin/clover/src/pipelines/gcp/funcs/actions/create.ts

```diff
@@ -1,11 +1,130 @@
 async function main(component: Input): Promise<Output> {
-  const serviceAccountJson = requestStorage.getEnv("GOOGLE_APPLICATION_CREDENTIALS_JSON");
+  // Check if resource already exists
+  const existingPayload = component.properties.resource?.payload;
+  if (existingPayload) {
+    return {
+      status: "error",
+      message: "Resource already exists",
+      payload: existingPayload,
+    };
+  }
+
+  // Get the generated code from code gen function
+  const codeString = component.properties.code?.["GCP Create Code Gen"]?.code;
+  if (!codeString) {
+    return {
+      status: "error",
+      message: "Could not find GCP Create Code Gen code for resource",
+    };
+  }
+
+  // Get method metadata from domain.extra
+  const insertMethodJson = _.get(
+    component.properties,
+    ["domain", "extra", "insertMethod"],
+    "",
+  );
+
+  if (!insertMethodJson) {
+    return {
+      status: "error",
+      message: "No insert method metadata found - this resource may not support creation",
+    };
+  }
+
+  const insertMethod = JSON.parse(insertMethodJson);
+  const baseUrl = _.get(component.properties, ["domain", "extra", "baseUrl"], "");
 
+  // Get authentication token
+  const serviceAccountJson = requestStorage.getEnv("GOOGLE_APPLICATION_CREDENTIALS_JSON");
   if (!serviceAccountJson) {
     throw new Error("Google Cloud Credential not found. Please ensure a Google Cloud Credential is attached to this component.");
   }
 
-  // Activate service account using stdin (most secure - no file on disk)
+  const { token, projectId } = await getAccessToken(serviceAccountJson);
+
+  // Build the URL by replacing path parameters
+  let url = `${baseUrl}${insertMethod.path}`;
+
+  // Replace path parameters with values from resource_value or domain
+  if (insertMethod.parameterOrder) {
+    for (const paramName of insertMethod.parameterOrder) {
+      let paramValue;
+
+      // Use extracted project_id for project parameter
+      if (paramName === "project") {
+        paramValue = projectId;
+      } else {
+        paramValue = _.get(component.properties, ["resource", "payload", paramName]) ||
```

Comment:
we should not be using the resource_value here. It bails out early if it finds one, so it should be empty anyhow